src/Runner.Listener/Runner.cs

```diff
@@ -477,7 +478,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     // todo: add retries https://github.com/github/actions-broker/issues/49
                                     var runServer = HostContext.CreateService<IRunServer>();
                                     await runServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
-                                    var jobMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId);
+                                    Pipelines.AgentJobRequestMessage jobMessage = null;
+
+                                    var remainingRetryLimitTime = TimeSpan.FromMinutes(5);
```

Comment:
NIT: just curious, why a 5 minute window for retrying instead of a number of retries?

Reply:
I disscussed limits with @ericsciple , so I just based on his thoughts. 
But I think that for errors that can be caused by servers\services instability it is good idea to rely on timeouts and retry several times during some timeframe, instead of rely on number of retries itself